File: kgparserSpringBoot/src/main/java/ru/klavogonki/kgparser/freemarker/HaulUtils.java
package ru.klavogonki.kgparser.freemarker;

import lombok.extern.log4j.Log4j2;

@Log4j2
public final class HaulUtils {

    // todo: use some library with these constants
    // see https://stackoverflow.com/questions/2442486/time-consts-in-java
    public static final int SECONDS_IN_MINUTE = 60;
    public static final int MINUTES_IN_HOUR = 60;
    public static final int SECONDS_IN_HOUR = SECONDS_IN_MINUTE * MINUTES_IN_HOUR;

    private HaulUtils() {
    }

    public static String format(Integer haul) {
        if (haul == null || haul < 0) {
            return "—";
        }

        int hours = haul / SECONDS_IN_HOUR;
        int minutes = (haul % SECONDS_IN_HOUR) / SECONDS_IN_MINUTE;
        int seconds = haul % SECONDS_IN_MINUTE;

        // todo: библиотека со склонением слов
        String result = String.format("%d ч. %d мин. %d сек.", hours, minutes, seconds);
        logger.debug("Haul {} converted to formatted string: {}", haul, result);
        return result;
    }
}

File: kgstatsSrv/src/main/kotlin/ru/klavogonki/statistics/freemarker/HaulUtils.kt
package ru.klavogonki.statistics.freemarker

import org.apache.logging.log4j.kotlin.Logging
import ru.klavogonki.statistics.util.DateUtils

object HaulUtils : Logging {

    const val NO_HAUL_STRING = "—"

    @JvmStatic
    fun format(haul: Int?): String {
        if (haul == null || haul < 0) {
            return NO_HAUL_STRING
        }

        val hours = haul / DateUtils.SECONDS_IN_HOUR
        val minutes = (haul % DateUtils.SECONDS_IN_HOUR) / DateUtils.SECONDS_IN_MINUTE
        val seconds = haul % DateUtils.SECONDS_IN_MINUTE

        val minutesFormatted = "%02d".format(minutes)
        val secondsFormatted = "%02d".format(seconds)

        // todo: библиотека со склонением слов
        val result = "$hours ч. $minutesFormatted мин. $secondsFormatted сек."
        logger.debug("Haul $haul converted to formatted string \"$result\".")

        return result
    }
}
